src/mds/MDCache.cc

```diff
@@ -511,6 +511,13 @@ void MDCache::_create_system_file(CDir *dir, std::string_view name, CInode *in,
   dout(10) << "_create_system_file " << name << " in " << *dir << dendl;
   CDentry *dn = dir->add_null_dentry(name);
 
+  // Pin the stray dentries until the open root is finished to
+  // prevent the stray dentries to be trimmed before pinned in
+  // the _create_system_file_finish().
+  if (in->is_stray()) {
+    dn->get(0);
```

Comment:
I guess we need to give this `0` pin a name:

````
2021-06-16T23:30:00.326 INFO:tasks.ceph.mds.c.smithi193.stderr:/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-5307-g06fb5cf0/rpm/el8/BUILD/ceph-17.0.0-5307-g06fb5cf0/src/mds/MDSCacheObject.cc: In function 'std::string_view MDSCacheObject::generic_pin_name(int) const' thread 7f5b6fce7700 time 2021-06-16T23:30:00.324579+0000
2021-06-16T23:30:00.326 INFO:tasks.ceph.mds.c.smithi193.stderr:/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/17.0.0-5307-g06fb5cf0/rpm/el8/BUILD/ceph-17.0.0-5307-g06fb5cf0/src/mds/MDSCacheObject.cc: 22: ceph_abort_msg("abort() called")
2021-06-16T23:30:00.326 INFO:tasks.ceph.mds.c.smithi193.stderr: ceph version 17.0.0-5307-g06fb5cf0 (06fb5cf0031e099ece537a86a27543dc4010ce0c) quincy (dev)
2021-06-16T23:30:00.327 INFO:tasks.ceph.mds.c.smithi193.stderr: 1: (ceph::__ceph_abort(char const*, int, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)+0xe5) [0x7f5b7cf4d73a]
2021-06-16T23:30:00.327 INFO:tasks.ceph.mds.c.smithi193.stderr: 2: (MDSCacheObject::generic_pin_name(int) const+0x1fe) [0x55765b07fe9e]
2021-06-16T23:30:00.327 INFO:tasks.ceph.mds.c.smithi193.stderr: 3: (operator<<(std::ostream&, CDentry const&)+0xc34) [0x55765af97824]
2021-06-16T23:30:00.327 INFO:tasks.ceph.mds.c.smithi193.stderr: 4: (CDentry::pre_dirty(unsigned long)+0x1fa) [0x55765af9899a]
2021-06-16T23:30:00.328 INFO:tasks.ceph.mds.c.smithi193.stderr: 5: (MDCache::_create_system_file(CDir*, std::basic_string_view<char, std::char_traits<char> >, CInode*, MDSContext*)+0x1a6) [0x55765ae53826]
2021-06-16T23:30:00.328 INFO:tasks.ceph.mds.c.smithi193.stderr: 6: (MDCache::populate_mydir()+0x806) [0x55765ae98476]
2021-06-16T23:30:00.328 INFO:tasks.ceph.mds.c.smithi193.stderr: 7: (MDCache::open_root()+0xd8) [0x55765ae98768]
2021-06-16T23:30:00.328 INFO:tasks.ceph.mds.c.smithi193.stderr: 8: (C_MDS_RetryOpenRoot::finish(int)+0x27) [0x55765af01d37]
2021-06-16T23:30:00.328 INFO:tasks.ceph.mds.c.smithi193.stderr: 9: (MDSContext::complete(int)+0x56) [0x55765b05f9a6]
2021-06-16T23:30:00.329 INFO:tasks.ceph.mds.c.smithi193.stderr: 10: (MDSRank::_advance_queues()+0x84) [0x55765ad5f494]
2021-06-16T23:30:00.329 INFO:tasks.ceph.mds.c.smithi193.stderr: 11: (MDSRank::ProgressThread::entry()+0xc5) [0x55765ad5fbb5]
2021-06-16T23:30:00.329 INFO:tasks.ceph.mds.c.smithi193.stderr: 12: (Thread::_entry_func(void*)+0xd) [0x7f5b7d043d2d]
2021-06-16T23:30:00.329 INFO:tasks.ceph.mds.c.smithi193.stderr: 13: /lib64/libpthread.so.0(+0x814a) [0x7f5b7bce814a]
2021-06-16T23:30:00.330 INFO:tasks.ceph.mds.c.smithi193.stderr: 14: clone()
````

From: /ceph/teuthology-archive/pdonnell-2021-06-16_21:26:55-fs-wip-pdonnell-testing-20210616.191804-distro-basic-smithi/6175689/teuthology.log

Reply:
Done.

At the same time I just made this pin ref more general in case there have other type dentries will potentially have the same issue in the future.